rfcs/0229-mnemos-technical-readiness-levels.md

```diff
@@ -0,0 +1,155 @@
+# RFC - MnemOS Technical Readiness Level
+
+MnemOS will for a long time have a wide variety of capabilities in a wide variety of different levels of "done".
+
+## Overview
+
+In order to better signal how far along an implementation or idea is, this RFC proposes establishing the following levels as somewhat of a standard:
+
+* A: Production Level - paid support available
+* B: Ready for general usage
+* C: In wide usage
+* D: In Implementation
+* E: In Design/RFC/Experiment
+* F: Early proposal
+* G-W: Reserved
+* X: Deprecated
+* Y: Retired
+* Z: Declined
```

Comment:
i'm not _convinced_ that letters are the best naming scheme here. they do indicate an ordering of states, which is valuable, and give the states a very short name, which is also potentially valuable. however, it's not super obvious what they mean if documentation just states that something has TRL "A" --- the reader has to then go look at the docs to see what TRL "A" means. it might be better to indicate that the _names_ of the TRLs should be used in documentation, rather than the letters, and the lettering/numbering scheme should be used primarily internally to this document?

so, for example, the docs for a component maybe should generally say:
> **Technical Readiness Level:** Production-Ready (A)

rather than
> **Technical Readiness Level:** A

what do you think?

Reply:
Yeah, I think there's something to be said about abbreviations being convenient for those in the know, and confusing for those that aren't.

We could also ditch the letters and give each level a punchy one word name to keep things easy to include and repeat.